### PR TITLE
Run Arm GNU cross-compiled tests in CI

### PR DESCRIPTION
**Pull request by keeranroth**
Add the running of cross-compiled GNU Arm builds to CI, running through QEMU. Changes are mad with a view of making things easier for future additions of cross-compilers and architectures:

* Install qemu-user-static in CI so that we can run aarch64 executables without needing to prefix qemu-aarch64 to the start of command lines
*  Rename toolchain file, so that it fits the pattern of `<arch>-<compiler>-crosscompile-toolchain.cmake`, so that we can easily include other compiler and arches without needing to update build scripts
* Pass the platform to run on into the test.sh script, and indicate when we are cross-compiling
* Export the CMAKE_PREFIX_PATH with the oneTBB install location prefixed, so that the test build can pick up the right TBB
* Print a message for failing tests. Currently the error output was not printed
* Examples CMake identifies the processor to run on using CMAKE_SYSTEM_PROCESSOR, instead of CMAKE_HOST_SYSTEM_PROCESSOR, which was leading to Intel being in directory names containing Arm ELFs
* Added files to list tests not to be run under emulation

### CODE REVIEW

**Comment by keeranroth** [comment]
This has the changes in https://github.com/oneapi-src/oneDAL/pull/2711 included. Put up for review now so that @rakshithgb-fujitsu can have a look at it, as he has very similar changes up. Worth aligning on one or the other implementation

File: .ci/env/tbb.sh [hunk 1]
```diff
@@ -59,69 +78,75 @@ done
...
-    pushd lnx
-        mkdir -p lib/${arch_dir}/gcc4.8
-        mv lib/libtbb* lib/${arch_dir}/gcc4.8
-    popd
-popd 
+echo cmake "${cmake_options[@]}"
+cmake "${cmake_options[@]}"
+make -j"$(nproc --all)"
+make install
+popd
```
Comment (Alexandr-Solovev) [inline .ci/env/tbb.sh:152]: Please, add a newline here

**Review by Alexandr-Solovev — COMMENTED** [review]
In general LGTM, lets wait CI checks

**Comment by Alexandr-Solovev** [comment]
Hi @keeranroth! Thanks for the pr! Ci is pretty green, only one check is red. The check is copyright. Please add copyrights in examples/oneapi/cpp/target_excludes.cmake and examples/daal/cpp/target_excludes.cmake after that it can be merged

File: examples/daal/cpp/target_excludes.cmake [hunk 2]
```diff
@@ -0,0 +1,34 @@
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
```
Comment (Alexandr-Solovev) [inline examples/daal/cpp/target_excludes.cmake:1]: Please add copyright

File: examples/oneapi/cpp/target_excludes.cmake [hunk 3]
```diff
@@ -0,0 +1,15 @@
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL CMAKE_HOST_SYSTEM_PROCESSOR)
```
Comment (Alexandr-Solovev) [inline examples/oneapi/cpp/target_excludes.cmake:1]: please add Copyright

**Comment by keeranroth** [comment]
@Alexandr-Solovev thanks for the reviews. I've added the copyright notices. Let me know if anything else needs updating

**Review by Alexandr-Solovev — APPROVED** [review]
Thanks! Ci is green!

**Review by napetrov — APPROVED** [review]

**Pull request merged**